src/components/NanoContract/NanoContractTransactionsList.component.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useNavigation } from '@react-navigation/native';
+import React, { useState } from 'react';
+import {
+  StyleSheet,
+  View,
+} from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { NanoContractTransactionsListHeader } from './NanoContractTransactionsListHeader.component';
+import { NanoContractTransactionsListItem } from './NanoContractTransactionsListItem.component';
+import { COLORS } from '../../styles/themes';
+import { nanoContractAddressChangeRequest } from '../../actions';
+import { HathorFlatList } from '../HathorFlatList';
+
+const getNanoContractHistory = (ncId) => (state) => {
+  const history = state.nanoContract.history[ncId];
+  return Object.values(history || {});
+}
+
+/**
+ * It presents a list of transactions from selected Nano Contract.
+ *
+ * @param {Object} props
+ * @param {Object} props.nc Nano Contract data
+ * @param {string} props.nc.ncId Nano Contract ID
+ * @param {string} props.nc.address Default caller address for Nano Contract interaction
+ */
+export const NanoContractTransactionsList = ({ nc }) => {
+  const ncHistory = useSelector(getNanoContractHistory(nc.ncId));
+
+  const dispatch = useDispatch();
+  const [ncAddress, changeNcAddress] = useState(nc.address);
```

Comment:
Why do you need to store this address in the component state if you can always get it from the redux?

Reply:
I don't remember why I did this way, but in this way it keeps the cycles segregated. Well, I'll try to use only the redux. Usually when the cycles are segregated the UI responds quicker...

Reply:
I tested it, it is to be this way indeed.